tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,338 @@ use strict;
 use base "console_yasttest";
 use testapi;
 use utils;
-use version_utils qw(is_sle is_leap);
-
-sub setup_ldap {
-    select_console 'root-console';
-    # workaround kernel message floating over console
-    assert_script_run "dmesg -n 4";
-
-    # check network at first
-    assert_script_run("if ! systemctl -q is-active network; then systemctl -q start network; fi");
-    # install local ldap server
-
-    if (is_sle '15+') {
-        my $ret = zypper_call('in yast2-auth-server openldap2-client tdb-tools', exitcode => [0, 104], timeout => 240);
-        return record_soft_failure 'bsc#1060870' if $ret == 104;
+use version_utils qw(is_sle is_leap is_tumbleweed);
+
+my %ldap_directives = (
+    fqdn                => 'openqa.ldap.test',
+    dir_instance        => 'openqa',
+    dir_suffix          => 'dc=suse,dc=de',
+    dn_container        => 'ou=testers,dc=suse,dc=de',
+    dir_manager_dn      => 'cn=qatester',
+    dir_manager_passwd  => 'openqatest',
+    ca_cert_pem         => '/etc/certificate.pem',
+    srv_cert_key_pkcs12 => '/etc/certificate.p12'
+);
+
+my %samba_directives = (
+    workgroup              => 'QA-Workgroup',
+    comment                => 'html docs for share',
+    path                   => '/home/html_public',
+    usershare_max_shares   => '90',
+    usershare_allow_guests => 'Yes',
+    netbios_name           => 'QA-Samba',
+    logon_drive            => 'C:',
+    wins_support           => 'Yes',
+    inherit_acls           => 'Yes',
+    read_only              => 'Yes'
+);
+
+sub smb_conf_checker {
+    my $error = "";
+    # select global & add share sections
+    my $select_script = q@
```

Comment:
Potential problem I see here is that we fail typing things once in a while. So there are multiple options to overcome it (assuming we run these tests on 64bit only and not some complex setup), put it in the data directory and download in SUT. Or we also can upload file content to the worker, so it can be downloaded in SUT (see http://open.qa/api/testapi/#_save_tmp_file).

Reply:
I have never noticed that we have a specific data directory in the test directory tree for this purpose. Thanks! But does not it take more resources to load another remote file, open & write its content to current console and execute it ? Sounds like a potential problem anyway, however I have no experience with it at all.

Reply:
Definitely it's not ideal way to execute scripts, but we don't have many options and typing long strings has shown us it's instability. As an alternative we can trigger whole test suite to run 100 times and see if it fails. So for all mentioned problems, it's just a feeling I got after fixing similar issues and not because it's best solution.

Reply:
I am just trying to implement in this way firewalld service workarounds for nis test and without any discussion it is good feature that I have never think about. I am glad that you are constantly sharing your experience with me, so let's do it using data dir. I have just not seen the benefit from it at first glance. :D Thanks for the explain

Reply:
You also have quite some experience with keymap and locale test suite, so it's always learnign the hard way =D